### ISSUE TITLE
Approach for 'negative' permissions testing?

### ISSUE THREAD

**Issue by dbinetti** [ydaniv/django-rest-assured#22]
First, awesome library.  Going to save me soooooo much time....

Second, what is the pattern for "negative" testing?  Meaning, say I have an API that is public-read, private-write: (using, say the `IsAuthenticatedOrReadOnly` permissions for `django-rest-framework`)  

I can test that anonymous users are able to Read, and I can test that authenticated users can Read or Write.  But what is the approach to ensure that anonymous users CAN NOT Write?  Obviously if I send anonymous users through the `ReadWrite` mixin it will fail (like in the tutorial), but that's not what i'm looking for:  I'm looking for a test that affirmatively asserts the negative permissions (if that makes any sense at all.... ) :-)

Basically, how do I write: assertEqual([anonymous user tries to PUT], [HTTP_401_UNAUTHORIZED])

Thanks...

**Comment by ydaniv** [comment]
Thank you so much, @dbinetti! Always great to hear (:

This makes a lot of sense, and I  sometimes test such "faulty" requests.
Your first option is to simply override the test method, say `test_update` for example, and do something as follows:

```
def test_update(self, data=None, results=None, use_patch=None, **kwargs):
        response = self.get_update_response(data, results, use_patch, **kwargs)
        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED, response.data)
```

Your second option is to offer a more generic way to accomplish that in a declarative form.

**Comment by ydaniv** [comment]
ping @dbinetti , assuming you figured your way out, so closing this.
If there's still something that I can do to help or you think is best changed here please feel free to comment/reopen.
Thanks!
